Extensions/XEP-0166/XMPPJingleSDP.m

```diff
@@ -895,68 +936,72 @@ - (XMPPIQ *)CandidateToXMPP:(NSDictionary *)dict media:(NSString *)media action:
-        [transElement addAttributeWithName:@"ufrag" stringValue:gUfrag];
-        [transElement addAttributeWithName:@"pwd" stringValue:gPwd];
-        
-        
-        NSXMLElement *canElement = [NSXMLElement elementWithName:@"candidate"];
-        
-        NSString *foundation = [candidate objectForKey:@"foundation"];
-        NSString *component = [candidate objectForKey:@"component"];
-        NSString *protocol = [candidate objectForKey:@"protocol"];
-        NSString *priority = [candidate objectForKey:@"priority"];
-        NSString *ip = [candidate objectForKey:@"ip"];
-        NSString *port = [candidate objectForKey:@"port"];
-        NSString *type = [candidate objectForKey:@"type"];
-        NSString *generation = [candidate objectForKey:@"generation"];
-        NSString *network = [candidate objectForKey:@"network"];
-        NSString *did = [candidate objectForKey:@"id"];
-
-        [canElement addAttributeWithName:@"foundation" stringValue:foundation];
-        [canElement addAttributeWithName:@"component" stringValue:component];
-        [canElement addAttributeWithName:@"protocol" stringValue:protocol];
-        [canElement addAttributeWithName:@"priority" stringValue:priority];
-        [canElement addAttributeWithName:@"ip" stringValue:ip];
-        [canElement addAttributeWithName:@"port" stringValue:port];
-        [canElement addAttributeWithName:@"type" stringValue:type];
-        [canElement addAttributeWithName:@"generation" stringValue:generation];
-        [canElement addAttributeWithName:@"network" stringValue:network];
-        [canElement addAttributeWithName:@"id" stringValue:did];
-        
-        if ([type isEqualToString:@"srflx"] || [type isEqualToString:@"prflx"] || [type isEqualToString:@"relay"]) {
-            NSString *relAddr = candidate[@"relAddr"];
-            NSString *relPort = candidate[@"relPort"];
-            [canElement addAttributeWithName:@"rel-addr" stringValue:relAddr];
-            [canElement addAttributeWithName:@"rel-port" stringValue:relPort];
-        }
-        
-        [transElement addChild:canElement];
-        
-        [transElement addChild:[fprElement1 copy]];
-        
-        [contentElement addChild:transElement];
-        [jingleElement addChild:contentElement];
-
+    NSString *mediaId = media ?: [dict objectForKey:@"id"];
+    if (mediaId) {
+        [jingleElement addChild:[self parseCandidate:candidate forMedia:mediaId]];
     }
     
     xmpp  = [[XMPPIQ alloc]initWithType:@"set" to:target elementID:UID child:[jingleElement copy]];
     
     return xmpp;
 
 }
+
+- (NSXMLElement *)parseCandidate:(NSDictionary *)candidate forMedia:(NSString *)media {
+    
+    NSXMLElement *contentElement = [NSXMLElement elementWithName:@"content"];
+    [contentElement addAttributeWithName:@"creator" stringValue:@"responder"];
+    [contentElement addAttributeWithName:@"name" stringValue:media];
+    
+    //NSXMLElement *transElement = transElement1;
```

Comment:
@igorcferreira can we remove this or is it useful?

Reply:
@fwroberto Done